### PR TITLE
feat: Added "make lint" target and added to "make test" target 

### PR DESCRIPTION
**Pull request by marcpfuller**
If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)  N/A
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?) Did an integration test details below
- [ ] I have opened a PR for the related docs change (if not, why?) N/A

## Testing Instructions

Ran edgex-go using local go-mod-bootstrap in both security and no-secty to make sure everything came up and no errors in logs
## New Dependency Instructions (If applicable)
N/A

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 tidy:
 	go mod tidy
 
-test:
+lint:
+	@which golangci-lint >/dev/null || echo "WARNING: go linter not installed. To install, run\n  curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b \$$(go env GOPATH)/bin v1.42.1"
+	@if [ "z${ARCH}" = "zx86_64" ] && which golangci-lint >/dev/null ; then golangci-lint run --config .golangci.yml ; else echo "WARNING: Linting skipped (not on x86_64 or linter not installed)"; fi
+
+
+test: lint
```
Comment (lenny-intel) [inline Makefile:14]: Restructure this as @bnevis-i did for this PR: https://github.com/edgexfoundry/go-mod-configuration/pull/57/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52

File: bootstrap/bootstrap.go [hunk 2]
```diff
@@ -155,7 +155,8 @@ func RunAndReturnWaitGroup(
 	// call individual bootstrap handlers.
 	startedSuccessfully := true
 	for i := range handlers {
-		if handlers[i](ctx, &wg, startupTimer, dic) == false {
+		//if handlers[i](ctx, &wg, startupTimer, dic) == false {
```
Comment (lenny-intel) [inline bootstrap/bootstrap.go:158]: Remove commented out code

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by bnevis-i — APPROVED** [review]
LGTM.

One optional change that I've made to the other Makefiles is to break out "go test" into a "unittest" target.

That way, the unit tests (the most important thing) run first, then the linter, then vet, gofmt, etc.

File: bootstrap/bootstrap.go (hunk 2 above)
Comment (marcpfuller) [reply bootstrap/bootstrap.go:158]: Commented code has been removed

File: Makefile (hunk 1 above)
Comment (marcpfuller) [reply Makefile:14]: I have restructured the Makefile per your suggestion

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**